### PR TITLE
TW-893: Remove pill on mentionned users

### PR DESCRIPTION
**Pull request by imGok**
- New mentionned pill in text message (blue text)
- Matrix ID in suggestion panel

image
image
image

Case tested : 
- long display name
- tag with message
- tag with message for long long display name
- tag with message for long long display name with line break in display name
- tag with very short display name 
- tag with icon inside name

https://github.com/linagora/twake-on-matrix/assets/48354990/7b4129ab-4307-4258-b148-908abd7d5766

https://github.com/linagora/twake-on-matrix/assets/48354990/ca0609eb-ad4e-4b6b-94b4-76c1764abb00

#893

### CODE REVIEW

File: lib/widgets/pill.dart [hunk 1]
```diff
@@ -27,48 +26,18 @@ class Pill extends StatelessWidget {
...
-            ),
-          ],
-        ),
-      ),
+    return GestureDetector(
       onTap: () {
         onTap?.call(url);
       },
+      child: Text(
+        displayName.shortenDisplayName(
```
Comment (nqhhdev) [inline lib/widgets/pill.dart:34]: Should display `matrixId`. 
Ex: matrixId is `@alice:domain-name` => `@alice`

**Comment by nqhhdev** [comment]
Pls demo case tag in input bar

File: lib/widgets/pill.dart [hunk 2]
```diff
@@ -27,48 +26,18 @@ class Pill extends StatelessWidget {
...
+          maxCharacters: maxCharactersDisplayNameForPill,
```
Comment (nqhhdev) [inline lib/widgets/pill.dart:35]: If you didn’t add the person to your address book we display their name as they named themselves during sign-up (we limit the character count by 12 symbols excl. spaces.).

Pls update it

**Comment by hoangdat** [comment]
> Pls demo case tag in input bar
> 
> Screenshot 2023-11-08 at 08 35 41

I prefer to keep the performance for composer than make it like the design for this case. Please consider it carefully

File: lib/widgets/mentionned_user.dart [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  const MentionnedUser({
+    Key? key,
+    required this.identifier,
+    required this.chatController,
+    required this.url,
+    this.textStyle,
+    this.onTap,
+  }) : super(key: key);
+
+  static const int maxCharactersDisplayNameForPill = 12;
```
Comment (nqhhdev) [inline lib/widgets/mentionned_user.dart:23]: make it private

File: lib/pages/chat/input_bar/input_bar.dart [hunk 4]
```diff
@@ -548,12 +549,34 @@ class SuggestionTile extends StatelessWidget {
...
+                mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                children: [
+                  Flexible(
+                    child: Text(
+                      suggestion['displayname'] ?? suggestion['mxid']!,
+                      maxLines: 1,
+                      overflow: TextOverflow.ellipsis,
+                      style: Theme.of(context).textTheme.bodyLarge?.copyWith(
+                            color: Theme.of(context).colorScheme.onSurface,
+                            fontSize: 17,
```
Comment (nqhhdev) [inline lib/pages/chat/input_bar/input_bar.dart:565]: no need

File: lib/pages/chat/input_bar/input_bar.dart [hunk 5]
```diff
@@ -535,7 +536,7 @@ class SuggestionTile extends StatelessWidget {
     if (suggestion['type'] == 'user' || suggestion['type'] == 'room') {
       final url = Uri.parse(suggestion['avatar_url'] ?? '');
       return Container(
-        padding: const EdgeInsetsDirectional.all(4.0),
+        padding: const EdgeInsetsDirectional.all(8.0),
```
Comment (nqhhdev) [inline lib/pages/chat/input_bar/input_bar.dart:539]: move it to style

File: lib/widgets/mentionned_user.dart [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+import 'package:fluffychat/utils/string_extension.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_matrix_html/text_parser.dart';
+
+class MentionnedUser extends StatelessWidget {
+  final String identifier;
+  final String url;
+  final OnPillTap? onTap;
+  final TextStyle? textStyle;
+  final ChatController chatController;
```
Comment (sherlockvn) [inline lib/widgets/mentionned_user.dart:12]: remove `ChatController`, i think `User user` is enough

Comment (imGok) [reply lib/widgets/mentionned_user.dart]: done, I simplified the widget

**Review by sherlockvn — APPROVED** [review]

**Comment by hoangdat** [comment]
for this case: 
image
what happens when user try to click on tag?

**Comment by hoangdat** [comment]
Please add and the case @[Hoang D 👍]

**Comment by imGok** [comment]
> for this case: image what happens when user try to click on tag?

it opens the user details (web demo video)
image

**Comment by imGok** [comment]
https://github.com/linagora/twake-on-matrix/assets/48354990/46cd1c64-d015-4888-be9d-d512182dc177

**Comment by imGok** [comment]
@hoangdat 

image

https://github.com/linagora/twake-on-matrix/assets/48354990/5cf5cf6f-eb73-4345-9ad1-ec9a0a232c0f

**Pull request merged**